### ISSUE TITLE
`VelloScene` doesn't respect Bevy `RenderLayers`

### ISSUE THREAD

**Issue by dannymcgee** [linebender/bevy_vello#58]
I'm experimenting with Vello in a Bevy 3D project, where I have my cameras set up like this:

```rs
fn spawn_cameras(mut cmd: Commands) {
    cmd.spawn((
        Name::new("Main Camera"),
        MainCamera,
        Camera3dBundle::default(),
    ));
    cmd.spawn((
        Name::new("UI Camera"),
        UiCamera,
        Camera2dBundle {
            camera: Camera {
                clear_color: ClearColorConfig::None,
                order: 1,
                ..default()
            },
            ..default()
        },
        RenderLayers::layer(1),
    ));
}
```

With the intention being that anything that needs to be rendered in 2D-world-space as an overlay on top of the 3D scene should get `RenderLayers::layer(1)`, so that it _only_ gets rendered by the 2D camera.

However, when I tried to spawn a `VelloSceneBundle` with a `RenderLayers::layer(1)` component, it didn't render at all. After some debugging, I discovered that after changing the camera's render layers to `RenderLayers::layer(0).with(1)`, the `VelloScene` started rendering as expected, indicating that `bevy_vello`'s rendering systems are ignoring the `RenderLayers` component.

This is problematic for me, because allowing my 2D camera to render layer 0 causes it to render some other things that it shouldn't (e.g., 3D-world-space Gizmos with an always-in-front depth bias).

Edit: I just read over the discussion in #49, so it looks like you're already aware of the issue. 🙂 Let me know if there's anything I can do to help!

**Comment by dannymcgee** [comment]
So it looks like Bevy UI _also_ ignores `RenderLayers`. 😂 Upon further inspection, there's actually a `TargetCamera` component in Bevy UI that serves this purpose. That doesn't fix my issue with non-UI stuff rendering to the UI camera, but I guess I'll have to come up with another solution for that. Closing this since it matches Bevy's behavior.

**Comment by simbleau** [comment]
This is definitely something I'd love to accept in a PR, or get to as time permits.

**Comment by simbleau** [comment]
Up to this point, `RenderLayer` was something I've never worked with. Currently bevy_vello ignores `RenderLayers` entirely, so at first it was unclear why adding a `RenderLayer::layer(1)` component to a `VelloSceneBundle` would make it not render.

I understand this a little better now and I'm committed to pushing this across the finish line.

Here's what I understand, for others:
- Changing the camera order will change the ordering of rendering cameras. This is a possible way to make bevy_vello render on top or on bottom of your 3d space. (camera order 1 or -1 respectively should do the trick).
![image](https://github.com/user-attachments/assets/85391ec7-6a74-4562-a629-718b3984486e)
- Camera ordering isn't enough because UI, gizmos, bevy_vello, etc. can and will use the same 2d camera. It's also sometimes desirable to say, for example, "I want everything to render above terrain." Hence, `RenderLayers` is also a way to sort by depth. See https://github.com/bevyengine/bevy/issues/1275 for more conversation. 
 
So there's two considerations that I believe are needing implementation:
- The situation where a camera has a `RenderLayer`. This would be useful to solve this issue with Gizmos. However I believe having camera with a different `RenderLayer` would mean internally we would need a new framebuffer for every `RenderLayer` the user wants. Vello is a one-pass renderer, which means it efficiently encodes all applicable scenes at once to create the final image. We do z-ordering via the transform's Z-component, but all of that is within vello. There's no way to render a vello background, 3d sphere, and a vello foreground. That would require 2 texture for Vello to render to, which isn't possible currently with bevy_vello. This situation would require more thought, as it would mean we'd need to adjust the pipeline to synchronize more than 1 vello render pass every frame.
- The other situation to account for is if you can have an entity, e.g. `VelloSceneBundle` on a `RenderLayer`, as an alternative means of Z-sorting within bevy_vello. This would be that "I want everything above the terrain" example. This is useful, but it doesn't seem like the reason this issue was made. bevy_vello already z-sorts by the transform z-component. Hence, you could set the terrain's `VelloScene` transform z to `f32::MIN`. However, I do like the idea, and I would see the value in wanting to render in layers, e.g. Terrain -> Unit layer -> Sky layer.

**Comment by simbleau** [comment]
@dannymcgee Let me know if I'm offbase in my understanding here

**Comment by ChristopherBiscardi** [comment]
I'll add the reason I was looking into this as well:

I have a Camera2d where I want to render all ui from a bevy_vello powered ui crate, and a Camera3d where I want to render "everything else" (gltf scenes, gizmos, effects, etc. game stuff).

If the Camera2d doesn't have a RenderLayers specified, then it will render a bunch of "extra" stuff, such as gizmos I'm using in the 3d camera. So ideally I would be able to order +1 on Camera2d, which places it over the 3d camera, and use RenderLayers to target all vello content to the Camera2d. I believe this still fits the "one pass" model of vello as it is only one layer.

**Comment by simbleau** [comment]
Fixed via #68  - Please re-open a new issue if there are further render layer issues.

There is also a new example for render layers.
